pkg/authentication/authentication.go

```diff
@@ -192,9 +218,16 @@ func (a *Handler) handleTokenExchange(w http.ResponseWriter, r *http.Request) {
 	}
 
 	accountProviders, _ := dbUser.GetAccountProviders()
+
+	// Extract primary email and avatar from account providers
+	primaryEmail := getPrimaryEmail(accountProviders)
+	primaryAvatar := getPrimaryAvatar(accountProviders)
+
 	authUser := User{
 		ID:               dbUser.ID.String(),
 		Name:             dbUser.Name,
+		Email:            primaryEmail,
```

Comment:
No need to use a `primaryEmail` variable, just use getPrimaryEmail(accountProviders) directly here